packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -747,6 +748,18 @@ mixin WidgetInspectorService {
   /// The current [WidgetInspectorService].
   static WidgetInspectorService get instance => _instance;
   static WidgetInspectorService _instance = _WidgetInspectorService();
+
+  final ValueNotifier<bool> _isSelectMode = ValueNotifier<bool>(true);
```

Comment:
this is a good refactor. A similar refactor is needed for _selectionChangedCallback and the selection object. Right now, only one of the screens is going to update on selection changed because there is a single listener instead of a value notifier.